nlmod/read/regis.py

```diff
@@ -228,13 +243,21 @@ def add_geotop_to_regis_layers(
                 geotop_k = geotop.get_lithok_props()
             gt = geotop.add_kh_and_kv(gt, geotop_k, anisotropy=anisotropy)
 
+    # copy the regis-dataset, before altering it
+    rg = rg.copy(deep=True)
+    if "layer" in rg["top"].dims:
+        msg = "Top in rg has a layer dimension. add_geotop_to_regis_layers will remove the layer dimension from top in rg."
+        logger.warning(msg)
+    else:
+        # temporarily add layer dimension to top in rg
+        rg["top"] = rg["botm"] + calculate_thickness(rg)
```

Comment:
instead of replacing the 2d top we want, can we just introduce a temporary variable?